src/overlays/actors/ovl_En_Tg/z_en_tg.c

```diff
@@ -76,37 +91,193 @@ static DamageTable D_809901F8 = {
+                     animationInfo->mode, animationInfo->morphFrames);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tg/func_8098F8A8.s")
+void EnTg_UpdateCollider(EnTg* this, PlayState* play) {
+    this->collider.dim.pos.x = this->actor.world.pos.x;
+    this->collider.dim.pos.y = this->actor.world.pos.y;
+    this->collider.dim.pos.z = this->actor.world.pos.z;
+    CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tg/func_8098F928.s")
+void EnTg_UpdateSkelAnime(EnTg* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tg/EnTg_Init.s")
+void EnTg_Init(Actor* thisx, PlayState* play) {
+    EnTg* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tg/EnTg_Destroy.s")
+    ActorShape_Init(&this->actor.shape, 0.0f, NULL, 0.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gHoneyAndDarlingSkel, NULL, this->jointTable, this->morphTable,
+                       HONEY_AND_DARLING_LIMB_MAX);
+    EnTg_ChangeAnim(&this->skelAnime, &sAnimationInfo, 0);
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = EnTg_Idle;
+    this->actor.gravity = -4.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tg/func_8098FA70.s")
+void EnTg_Destroy(Actor* thisx, PlayState* play) {
+    EnTg* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tg/EnTg_Update.s")
+    Collider_DestroyCylinder(play, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tg/func_8098FBB4.s")
+/**
+ * The actor spins, and a heart is spawned above the actor every 12 frames.
+ */
+void EnTg_Idle(EnTg* this, PlayState* play) {
+    Vec3f heartStartPos;
+
+    this->actor.shape.rot.y += sREG(0) + 0x258;
+    this->actor.world.rot = this->actor.shape.rot;
+
+    if (DECR(this->spawnHeartTimer) == 0) {
+        this->spawnHeartTimer = 12;
+        heartStartPos = this->actor.world.pos;
+        heartStartPos.y += 62.0f;
+        EnTg_SpawnHeart(this, this->enTgHeartEffect, &heartStartPos, ARRAY_COUNT(this->enTgHeartEffect));
+    }
+}
+
+void EnTg_Update(Actor* thisx, PlayState* play) {
+    EnTg* this = THIS;
+
+    this->actionFunc(this, play);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 0.0f, 0.0f, 0.0f, 4U);
+    EnTg_UpdateSkelAnime(this, play);
+    EnTg_UpdateHearts(play, this->enTgHeartEffect, ARRAY_COUNT(this->enTgHeartEffect));
+    EnTg_UpdateCollider(this, play);
+}
+
+s32 EnTg_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    return 0;
```

Comment:
This function returns a boolean:
```suggestion
    return false;
```